app/models/post.rb

```diff
@@ -10,7 +10,7 @@ class Post < ApplicationRecord
   validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
   def last_five_comments
-    comments.order(created_at: :desc).limit(5)
+    comments.includes(:author).order(created_at: :desc).limit(5)
```

Comment:
- [ ] You have done a great job trying to resolve the N + 1 problem for fetching posts. However, to really make sure that this is fully resolved and meet all the project requirements **(Note:** N+1 problem is solved when fetching all posts and their comments for a user.) )you have to make this change as well in the index action in your app/controllers/posts_controller.rb file as this action is the one responsible for fetching all the posts.
- [ ] To customize it to fetch all the posts for a user including the comments as well (Now fixing the N + 1 problem) You can easily do it as below:

```
def index
    @user = User.find(params[:user_id])
    @userposts = @user.posts.includes(:comments)
  end
```

![image](https://user-images.githubusercontent.com/86318284/205819939-4a96e29a-00ec-4a1e-ae71-802547e0d826.png)

Reply:
@Nemwel-Boniface Thank you so much for your advice! We are already working on it.

Reply:
I have finished applying your recommendation and the objective has become much clearer. Thank you very much!